Doc/whatsnew/3.14.rst

```diff
@@ -208,6 +209,30 @@ configuration mechanisms).
 .. seealso::
    :pep:`741`.
 
+.. _whatsnew314-tail-call:
+
+A new tail-calling interpreter
+------------------------------
+
+A new type of interpreter based on tail calls has been added to CPython.
+For certain newer compilers, this interpreter provides
+significantly better performance. Preliminary numbers on our machines suggest
+anywhere from -3% to 30% faster Python code, and a geometric mean of 9-15%
+faster on ``pyperformance`` depending on platform and architecture.
+
+This interpreter currently only works with Clang 19 and newer
+on x86-64 and AArch64 architectures. However, we expect
+that a future release of GCC will support this as well.
+
+This feature is opt-in for now. We highly recommend enabling profile-guided
+optimization with the new interpreter as it is the only configuration we have
+tested and can validate its improved performance.
+For further information on how to build Python, please see
```

Comment:
Usually no need for "please" in technical documentation ([Google](https://developers.google.com/style/word-list#please), [Microsoft](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/please), [Apple](https://support.apple.com/en-gb/guide/applestyleguide/apsg4473eab0/web))

```suggestion
For further information on how to build Python, see
```